tests/ui/traverse_bare_self.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0308]: mismatched types
+ --> tests/ui/traverse_bare_self.rs:8:6
+  |
+7 |   #[pymethods]
+  |   ------------ arguments to this function are incorrect
+8 |   impl TraverseTriesToTakePyRef {
+  |  ______^
+9 | |     fn __traverse__(slf: PyRef<Self>, visit: PyVisit) {}
+  | |___________________^ expected fn pointer, found fn item
+  |
+  = note: expected fn pointer `for<'a, 'b> fn(&'a TraverseTriesToTakePyRef, PyVisit<'b>) -> Result<(), PyTraverseError>`
+                found fn item `for<'a, 'b> fn(pyo3::PyRef<'a, TraverseTriesToTakePyRef>, PyVisit<'b>) {TraverseTriesToTakePyRef::__traverse__}`
+note: function defined here
+ --> src/impl_/pymethods.rs
+  |
+  | pub unsafe fn call_traverse_impl<T>(
+  |               ^^^^^^^^^^^^^^^^^^
```

Comment:
To make this error nicer we could make the proc-macro check that the `__traverse__` implementation has `&self` receiver? Happy to leave this as an issue for a follow-up in another PR.

Reply:
Due to my innate love for proc-macros, I would be glad if we could move this into a follow-up issue so it can get the proper attention it so rightfully deserves. ;)

Reply:
Hehe absolutely 🤣

Reply:
That issue is #3180